src/Microsoft.Performance.Toolkit.Engine/EngineExecutionResults.cs

```diff
@@ -215,5 +223,222 @@ public bool TryQueryOutput<T>(DataOutputPath dataOutputPath, out T data)
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        ///     Builds a table with the given <paramref name="tableDescriptor"/>.
+        /// </summary>
+        /// <param name="tableDescriptor">
+        ///     The table to build.
+        /// </param>
+        /// <returns>
+        ///     The built table.
+        /// </returns>
+        /// <exception cref="TableNotBuiltException">
+        ///     A table cannot be built for the given <paramref name="tableDescriptor"/>.
+        /// </exception>
+        public ITableResult BuildTable(TableDescriptor tableDescriptor)
+        {
+            if (TryBuildTable(tableDescriptor, out ITableResult filledTableBuilder))
+            {
+                return filledTableBuilder;
+            }
+
+            throw new TableNotBuiltException(tableDescriptor);
+        }
+
+        /// <summary>
+        ///     Attempts to build a table with the given <paramref name="tableDescriptor"/>.
+        /// </summary>
+        /// <param name="tableDescriptor">
+        ///     The table to build.
+        /// </param>
+        /// <param name="filledTableBulder">
+        ///     The built table, if any.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the table for the given <paramref name="tableDescriptor"/> was built;
+        ///     <c>false</c> otherwise.
+        /// </returns>
+        public bool TryBuildTable(TableDescriptor tableDescriptor, out ITableResult filledTableBulder)
+        {
+            var tableBuilder = new TableBuilder();
+
+            if (TryOnRepository(tableDescriptor, (reference, tableRetrieval) => { reference.BuildTableAction(tableBuilder, tableRetrieval); return tableBuilder; }, out ITableResult repoTableResult))
+            {
+                filledTableBulder = repoTableResult;
+                return true;
+            }
+
+            if (TryOnProcessor(tableDescriptor, (processor) => { processor.BuildTable(tableDescriptor, tableBuilder); return tableBuilder; }, out ITableResult processorTableResult))
+            {
+                filledTableBulder = processorTableResult;
+                return true;
+            }
+
+            filledTableBulder = null;
+            return false;
+        }
+
+        private bool TryOnRepository<TResult>(TableDescriptor tableDescriptor, Func<ITableExtensionReference, IDataExtensionRetrieval, TResult> func, out TResult result)
+        {
+            result = default;
+
+            if (this.repository.TablesById.TryGetValue(tableDescriptor.Guid, out ITableExtensionReference reference))
+            {
+                try
+                {
+                    var tableRetrieval = this.retrievalFactory.CreateDataRetrievalForTable(tableDescriptor.Guid);
+
+                    result = func(reference, tableRetrieval);                    
+
+                    return true;
+                }
+                catch
+                {
+                    return false;
```

Comment:
should we report some error to the user in this case? (same in TryOnProcessor) if we want to , we might have to let the exception bubble up to where we know what the func is actually doing

Reply:
gah, just realized this is in a Try[Method]. not sure what the right thing to do here is

---
In reply to: [592041314](https://github.com/microsoft/microsoft-performance-toolkit-sdk/pull/56#discussion_r592041314) [](ancestors = 592041314)

Reply:
maybe just have a TryGetOn[Repository|Processor], and not pass in the func?

---
In reply to: [592041488](https://github.com/microsoft/microsoft-performance-toolkit-sdk/pull/56#discussion_r592041488) [](ancestors = 592041488,592041314)